pvlib/iotools/acis.py

```diff
@@ -0,0 +1,483 @@
+import requests
+import pandas as pd
+import numpy as np
+
+
+VARIABLE_MAP = {
+    # time series names
+    'pcpn': 'precipitation',
+    'maxt': 'temp_air_max',
+    'avgt': 'temp_air_average',
+    'obst': 'temp_air_observation',
+    'mint': 'temp_air_min',
+    'snow': 'snowfall',
+    'snwd': 'snowdepth',
+
+    # metadata names
+    'lat': 'latitude',
+    'lon': 'longitude',
+    'elev': 'altitude',
+}
+
+
+def _get_acis(start, end, params, map_variables, url, **kwargs):
+    """
+    generic helper for the public get_acis_X functions
+    """
+    params = {
+        # use pd.to_datetime so that strings (e.g. '2021-01-01') are accepted
+        'sdate': pd.to_datetime(start).strftime('%Y-%m-%d'),
+        'edate': pd.to_datetime(end).strftime('%Y-%m-%d'),
+        'output': 'json',
+        **params,  # endpoint-specific parameters
+    }
+    response = requests.post(url,
+                             json=params,
+                             headers={"Content-Type": "application/json"},
+                             **kwargs)
+    response.raise_for_status()
+    payload = response.json()
+
+    # somewhat inconveniently, the ACIS API tends to return errors as "valid"
+    # responses instead of using proper HTTP error codes:
+    if "error" in payload:
+        raise requests.HTTPError(payload['error'], response=response)
+
+    columns = ['date'] + [e['name'] for e in params['elems']]
+    df = pd.DataFrame(payload['data'], columns=columns)
+    df = df.set_index('date')
+    df.index = pd.to_datetime(df.index)
+    df.index.name = None
+
+    metadata = payload['meta']
+    # for StnData endpoint, unpack combination "ll" into lat, lon 
+    ll = metadata.pop('ll', None)
+    if ll:
+        metadata['lon'], metadata['lat'] = ll
+
+    try:
+        metadata['elev'] = metadata['elev'] * 0.3048  # feet to meters
+    except KeyError:
+        # some queries don't return elevation
+        pass
+
+    if map_variables:
+        df = df.rename(columns=VARIABLE_MAP)
+
+        for key in list(metadata.keys()):
+            if key in VARIABLE_MAP:
+                metadata[VARIABLE_MAP[key]] = metadata.pop(key)
+
+    return df, metadata
+
+
+def get_acis_prism(latitude, longitude, start, end, map_variables=True,
+                   url="https://data.rcc-acis.org/GridData", **kwargs):
+    """
+    Retrieve estimated daily precipitation and temperature data from PRISM
+    via the Applied Climate Information System (ACIS).
+
+    Geographical coverage: approximately -130° to -65° in longitude and
+    0° to 50° in latitude.
```

Comment:
```suggestion
    Geographical coverage: approximately 0° to 50° in latitude and
    -130° to -65° in longitude.
```

To most folks, it might be nice to state it a bit more explicit:

    Geographical coverage: US, Central America, and part of South America.
    Approximately 0° to 50° in latitude and -130° to -65° in longitude.

Same comment below.